### PR TITLE
Use docker credentials to pull and push with authenticated registries

### PR DESCRIPTION
**Pull request by davidthor**
## Description

This PR improves the OCI library to read from the docker configuration file (default location of `~/.docker/config.json`) to collect and utilize credentials to authenticate with private registries. You'll still need to log into the registry using `docker login` for now, but this PR will ensure that `arcctl push <component-tag>` will successfully push artifacts to registries like Dockerhub that you're logged in to.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Make sure you have a dockerhub account before you begin:

```sh
$ docker login
$ git clone git@github.com:architect-team/mailslurper.git && cd mailslurper
$ arcctl build . -t <account>/<repo>:latest
$ arcctl push <account>/<repo>:latest
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/commands/build.ts [hunk 1]
```diff
@@ -63,6 +64,8 @@ async function build_action(options: BuildOptions, context_file: string): Promis
       buildArgs.push(path.join(Deno.cwd(), context, build_options.context));
     }
 
+    console.log('Running: docker', ...buildArgs);
```
Comment (mueschm) [inline src/commands/build.ts:67]: Is this supposed to be here? It is possible a build arg would have sensitive information.

File: src/commands/build.ts [hunk 2]
```diff
@@ -86,8 +89,47 @@ async function build_action(options: BuildOptions, context_file: string): Promis
...
+      `
+        FROM alpine:latest
+        WORKDIR /app
+        COPY ./contents .
+        CMD ["sh", "-c", "cp -r ./* $TARGET_DIR"]
+      `,
+    );
+
+    // Publish the volume container
+    if (options.verbose) {
```
Comment (mueschm) [inline src/commands/build.ts:109]: Could a lot of this code be merged into one path? The code above is pretty streamlined where the only difference is that the verbose flag decides on the quiet flag.

File: src/oci/image-repository.ts [hunk 3]
```diff
@@ -180,6 +259,7 @@ export class ImageRepository<C extends any = any> {
         digest: res.headers.get('docker-content-digest')!,
       };
     } catch (err) {
+      console.error(err);
```
Comment (mueschm) [inline src/oci/image-repository.ts:262]: Extra line?

**Review by mueschm — CHANGES_REQUESTED** [review]
Overall looks really solid

File: src/commands/build.ts (hunk 1 above)
Comment (davidthor) [reply src/commands/build.ts]: Removed.

File: src/commands/build.ts (hunk 2 above)
Comment (davidthor) [reply src/commands/build.ts:112]: Consolidated into a helper function.

File: src/oci/image-repository.ts (hunk 3 above)
Comment (davidthor) [reply src/oci/image-repository.ts]: Removed.

**Pull request merged**